Modules/Module_Processing/Module_ASL/xASL_wrp_Quantify.m

```diff
@@ -187,70 +187,95 @@ function xASL_wrp_Quantify(x, PWI4D_Path, pathOutputCBF, M0Path, SliceGradientPa
+% We can safely do this, because we didn't use this recently
+
+IndexSetsAge = find(strcmpi(x.S.SetsName, 'age'));
+IndexSetsSex = find(strcmpi(x.S.SetsName, 'sex'));
+indexSetsHct = find(strcmpi(x.S.SetsName, 'hematocrit'));
+
 % a. We prioritize participants.tsv>Hematocrit (x.S.SetsID > x.Hematocrit)
-indexSetsName = find(strcmpi(x.S.SetsName, 'hematocrit'));
-if ~isempty(indexSetsName) && isfield(x, 'Hematocrit')
-    warning('Found hematocrit values in participants.tsv & in x.Hematocrit, using the first');
-    x.Hematocrit = x.S.SetsID(:,indexSetsName);
-elseif ~isempty(indexSetsName)
-    x.Hematocrit = x.S.SetsID(:,indexSetsName);
+if ~isempty(indexSetsHct)
+    x.Q.Hematocrit = x.S.SetsID(x.iSubjectSession, indexSetsHct);
+    fprintf('%s\n', 'Using hematocrit found in participants.tsv for blood T1 correction');
 end
 
-% We convert x.Hematocrit > x.hematocrit 
-% -> this accounts for the (backward compatibility) possibility
-% that the user provides hematocrit in dataPar.json
-if isfield(x, 'hematocrit') && isfield(x, 'Hematocrit')
-    warning('Two hematocrit fields, ignoring x.hematocrit');
-    x = rmfield(x, 'hematocrit');
-elseif isfield(x, 'hematocrit')
-    x.Hematocrit = x.hematocrit;
-	x = rmfield(x, 'hematocrit');
+if isfield(x, 'Hematocrit')
+    warning('x.Hematocrit detected, we ignore this, considering adding hct values to participants.tsv instead');
+end
+if isfield(x, 'hematocrit')
+    warning('x.hematocrit detected, we ignore this, considering adding hct values to participants.tsv instead');
 end
 
-% b. We model the expected hematocrit from age & sex
-% only if the Hematocrit was not specifically provided
-if ~isfield(x, 'Hematocrit')
-    IndexSetsAge = find(strcmpi(x.S.SetsName, 'age'));
-    IndexSetsSex = find(strcmpi(x.S.SetsName, 'sex'));
-    if ~isempty(IndexSetsAge) || ~isempty(IndexSetsSex)
-        fprintf('%s\n', 'Trying to infer hematocrit from age & sex');
-        
-        age = x.S.SetsID(:, IndexSetsAge);
-        
-        % Convert sex correctly
-        sex = x.S.SetsID(:, IndexSetsSex);
-        sexOptions = x.S.SetsOptions{:, IndexSetsSex};
-        sexN = nan(length(sex), 1); % currently we can only infer hct from male/female
-        % So anything that is not detected, will remain NaNs
-        for iOption=1:length(sexOptions)
-            if ~isempty(regexpi(sexOptions{iOption}, '^(male|m|man|men)$'))
-                sexN(sex==iOption) = 1;
-            elseif ~isempty(regexpi(sexOptions{iOption}, '^(female|f|woman|women)$'))
-                sexN(sex==iOption) = 2;
-            end
-        end
 
-        if sum(isnan(sexN))>0
-            warning('Unknown sex detected in participants.tsv, currently we can only infer hematocrit from male or female');
-            fprintf('%s\n', 'So in participants.tsv specify the words "male" and "female" only');
+% a2. Manage hematocrit usage parameter
+if isfield(x.modules.asl.bHct2BloodT1)
```

Comment:
You are checking existence of bHct2DBloodT1. But not checking existence of asl field.

This should be done already in the ASL module - setting it to [] if missing.